### ISSUE TITLE
[Bug]: Document Viewer does not display PDFs with Content-Disposition: attachment as header

### ISSUE THREAD

**Issue by guilhermep** [appsmithorg/appsmith#9746]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

In the Document Viewer, when linking a PDF with ` Content-Disposition: attachment` as a response header, the file is downloaded in the browser. It should be displayed on the screen instead. 

### Steps To Reproduce

1. Insert a Document Viewer widget
2. Link a PDF with ` Content-Disposition: attachment` as a response header.

### Environment

Production

### Version

Self-Hosted

**Comment by dilippitchika** [comment]
@guilhermep we wanted to understand the logic behind why the `Content-Disposition: attachment` header is being used in your pdf files. Based on this [MDN Link](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition) I see that it means we should bypass rendering and download the attachment.

I also went through a lot of pdf links available online and wasn't able to find the header in those, is this a specific behavior within your org? If so is there a reason for it? 

Any help from you would help us get to the root faster

Some sample links where I checked for response header Content-Disposition, which was not there
[Link 1](https://smallpdf.com/handle-widget#url=https://assets.ctfassets.net/l3l0sjr15nav/29D2yYGKlHNm0fB2YM1uW4/8e638080a0603252b1a50f35ae8762fd/Get_Started_With_Smallpdf.pdf) 
[Link 2](https://file-examples-com.github.io/uploads/2017/10/file-sample_150kB.pdf)

**Comment by guilhermep** [comment]
The file is being served this way by a third party service. There is no specific reason that I'm aware.

In the context of the browser it makes sense for the file to be downloaded. In a Document Viewer widget, however, there is no situation in which the user would want this behavior, in my opinion.

[Here is a sample PDF.](https://download.novapdf.com/download/samples/pdf-example-bookmarks.pdf)

**Comment by somangshu** [comment]
@techbhavin can you tell us if removing the `Content-Disposition` from header possible for us before loading it in the browser?

To my knowledge the iFrame would act like the browser and download the file, If however the client can control the headers before loading them, This problem can be solved!

Thanks for the input @guilhermep

**Comment by techbhavin** [comment]
@somangshu, Here we can't remove  `Content-Disposition` from the header.

But here we can change use office viewer instead of the default pdf viewer (without adding any new package). so it will fix our problem.

**Comment by somangshu** [comment]
Lets go for it @techbhavin

**Comment by techbhavin** [comment]
> @somangshu, Here we can't remove `Content-Disposition` from the header.
> 
> But here we can change use office viewer instead of the default pdf viewer (without adding any new package). so it will fix our problem.

this solution is not working

**Comment by somangshu** [comment]
After doing some research around this we found out that we cannot modify the header value unless we have an access to the headers, Right now we do not have a direct solution to this and henceforth deprioritising it

**Comment by rafaelferreirapt** [comment]
Hi, I was able to reproduce this issue. If the file is uploaded with Content-Disposition: attachment; and since it's not possible to define the `response-headers {content-disposition "inline"}` when we generate a signedUrl it will be attachment and it will download instead of view in the Document Viewer.
My suggestion is to enable one option for setting the response headers when signing URLs and with that we can set the content disposition as inline.

**Comment by dilippitchika** [comment]
@rafaelferreirapt are you referring to accessing response headers in s3? so that it can be set to `content-dispostion Inline` when you are calling any s3 query?

**Comment by rafaelferreirapt** [comment]
@dilippitchika yes, because currently I can't do that and the content-disposition is what has been set to the object. with that I can't use the document viewer because always downloads me the file (because by default I have the content-disposition set to that for the production application and use cases). For the dashboard, I would like to be able to set the content-disposition to inline when creating the signing urls with the List method, for example.

**Comment by dilippitchika** [comment]
Are you uploading the files within appsmith to s3 @rafaelferreirapt ?

**Comment by rafaelferreirapt** [comment]
Hi! @dilippitchika, the files in my use case are uploaded by the miio system (our application). This use case is invoicing PDF files. Since they must be downloaded by the miio application  user (our product) they have content-disposition attachment. But, for our costumer support they should only be viewed (content-dispostion Inline). When generating the signed URL using the list method it would be nice if I can set the headers content-dispostion to Inline.
